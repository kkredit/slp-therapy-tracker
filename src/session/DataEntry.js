import React from 'react'
import './Session.css'
// import { Student, STUDENTS_MIN, STUDENTS_MAX, GOALS_MIN, GOALS_MAX } from './Student.js'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Carousel from 'react-bootstrap/Carousel'
import Button from 'react-bootstrap/Button'

const GoalButton = (props) => {
  return (
    <Button variant="light" size="lg" className="goalButton" onClick={props.onClick}>
      <i className={props.faClass} />
    </Button>
  );
}

const GoalCarouselItem = (props) => {
  return (
    <Carousel.Item>
      <Container>
        <Row>
          <Col xs>
            <h4 className="goalNum"> Goal {props.goal.number + 1} </h4>
          </Col>
        </Row>
        <Row>
          <Col xs/>
          <Col xs={3}>
            <GoalButton faClass="fa fa-3x fa-check" />
          </Col>
          <Col xs={3}>
            <GoalButton faClass="fa fa-3x fa-circle-o" />
          </Col>
          <Col xs={3}>
            <GoalButton faClass="fa fa-3x fa-times" />
          </Col>
          <Col xs/>
        </Row>
      </Container>
    </Carousel.Item>
  );
}

class GoalsCarouselSet extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      goals: props.goals,
    };
  }

  render() {
    const goalCarouselItems = this.state.goals.map((goal) => {
      return (
        <GoalCarouselItem key={goal.key}
                          goal={goal} />
      );
    });

    return (
      <React.Fragment>
        {goalCarouselItems}
      </React.Fragment>
    );
  }
}

class StudentRow extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      student: props.student,
    };
  }

  render() {
    return (
      <Row>
        <Col xs={1}>
          <h4 className="studentName"> Student {this.state.student.number + 1} </h4>
        </Col>
        <Col xs={11}>
          <Carousel className="goalsCarousel"
                    interval={null}
                    indicators={false}
                    // nextIcon={<i className="fa fa-angle-right" aria-hidden="false"/>}
                    >
            <GoalsCarouselSet student={this.state.student} goals={this.state.student.goals} />
          </Carousel>
        </Col>
      </Row>
    );
  }
}

class StudentRows extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      students: props.students,
    };
  }

  render() {
    const studentRowEntries = this.state.students.map((student) => {
      return (
        <StudentRow key={student.number} student={student} />
      );
    });

    return (
      <React.Fragment>
        {studentRowEntries}
      </React.Fragment>
    );
  }
}

class DataEntry extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      active: props.active,
    };
  }

  static getDerivedStateFromProps(props) {
    return {active: props.active, students: props.students};
  }

  render() {
    if (!this.state.active) {
      return null;
    }
    else {
      return (
        <React.Fragment>
          <Container>
            <Row>
              <Col xs={1}>
                <h3 className="dataColHdr"> Students </h3>
              </Col>
              <Col xs={1} />
              <Col xs={9}>
                <h3 className="dataColHdr"> Goals </h3>
              </Col>
              <Col xs={1} />
            </Row>
            <StudentRows students={this.state.students} />
          </Container>

          <Container>
            <Row>
              <Col xs={1}>
                <h3 className="dataColHdr"> Students </h3>
              </Col>
              <Col xs={1} />
              <Col xs={9}>
                <h3 className="dataColHdr"> Goals </h3>
              </Col>
              <Col xs={1} />
            </Row>
            <Row>
              <Col xs={1}>
                <h4 className="studentName"> Student {this.state.students[0].number + 1} </h4>
              </Col>
              <Col xs={11}>
                <Carousel className="goalsCarousel"
                          interval={null}
                          indicators={false}
                          // nextIcon={<i className="fa fa-angle-right" aria-hidden="false"/>}
                          >
                  <Carousel.Item>
                    <Container>
                      <Row>
                        <Col xs>
                          <h4 className="goalNum"> Goal {this.state.students[0].goals[0].number + 1} </h4>
                        </Col>
                      </Row>
                      <Row>
                        <Col xs/>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-check" />
                        </Col>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-circle-o" />
                        </Col>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-times" />
                        </Col>
                        <Col xs/>
                      </Row>
                    </Container>
                  </Carousel.Item>
                  <Carousel.Item>
                    <Container>
                      <Row>
                        <Col xs>
                          <h4 className="goalNum"> Goal {this.state.students[0].goals[1].number + 1} </h4>
                        </Col>
                      </Row>
                      <Row>
                        <Col xs/>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-check" />
                        </Col>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-circle-o" />
                        </Col>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-times" />
                        </Col>
                        <Col xs/>
                      </Row>
                    </Container>
                  </Carousel.Item>
                </Carousel>
              </Col>
            </Row>
            <Row>
              <Col xs={1}>
                <h4 className="studentName"> Student {this.state.students[1].number + 1} </h4>
              </Col>
              <Col xs={11}>
                <Carousel className="goalsCarousel"
                          interval={null}
                          indicators={false}
                          // nextIcon={<i className="fa fa-angle-right" aria-hidden="false"/>}
                          >
                  <Carousel.Item>
                    <Container>
                      <Row>
                        <Col xs>
                          <h4 className="goalNum"> Goal {this.state.students[1].goals[0].number + 1} </h4>
                        </Col>
                      </Row>
                      <Row>
                        <Col xs/>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-check" />
                        </Col>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-circle-o" />
                        </Col>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-times" />
                        </Col>
                        <Col xs/>
                      </Row>
                    </Container>
                  </Carousel.Item>
                  <Carousel.Item>
                    <Container>
                      <Row>
                        <Col xs>
                          <h4 className="goalNum"> Goal {this.state.students[1].goals[1].number + 1} </h4>
                        </Col>
                      </Row>
                      <Row>
                        <Col xs/>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-check" />
                        </Col>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-circle-o" />
                        </Col>
                        <Col xs={3}>
                          <GoalButton faClass="fa fa-3x fa-times" />
                        </Col>
                        <Col xs/>
                      </Row>
                    </Container>
                  </Carousel.Item>
                </Carousel>
              </Col>
            </Row>
          </Container>
        </React.Fragment>
      );
    }
  }
}

export default DataEntry;
