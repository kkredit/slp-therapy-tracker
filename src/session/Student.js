
export const STUDENTS_MIN = 1;
export const STUDENTS_MAX = 4;
export const GOALS_MIN = 1;
export const GOALS_MAX = 4;

export const ATTEMPT_SUCC = 0;
export const ATTEMPT_FAIL = 1;
export const ATTEMPT_CUED = 2;

export class Goal {
  constructor(number) {
    this.key = number;
    this.number = number;
    this.attempts = [];

    this.addAttempt = this.addAttempt.bind(this);
    this.addSuccess = this.addSuccess.bind(this);
    this.addFail = this.addFail.bind(this);
    this.addCued = this.addCued.bind(this);
  }

  addAttempt(type) {
    this.attempts.push(type);
  }

  addSuccess() {
    this.addAttempt(ATTEMPT_SUCC);
  }

  addFail() {
    this.addAttempt(ATTEMPT_FAIL);
  }

  addCued() {
    this.addAttempt(ATTEMPT_CUED);
  }
}

export class Student {
  constructor(number, updateCb) {
    this.key = number;
    this.number = number;
    this.goals = [new Goal(0)];
    this.updateCb = updateCb;

    this.addGoal = this.addGoal.bind(this);
    this.delGoal = this.delGoal.bind(this);
  }

  addGoal() {
    if (this.goals.length < GOALS_MAX) {
      this.goals.push(new Goal(this.goals.length));
      this.updateCb();
    }
  }

  delGoal() {
    if (this.goals.length > GOALS_MIN) {
      this.goals.pop();
      this.updateCb();
    }
  }
}
