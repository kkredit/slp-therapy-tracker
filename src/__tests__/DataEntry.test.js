import React from 'react';
import ReactDOM from 'react-dom';
import DataEntry from '../session/DataEntry.js'
import { Student, Goal } from '../session/Student.js'
import GetTestDb from '../__test_helpers__/SetupDb.js'
import { create } from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

it('renders without crashing', () => {
  const div = document.createElement('div');
  let db = GetTestDb();
  ReactDOM.render(
    <BrowserRouter>
      <DataEntry active={true}
                 students={[new Student(0, null)]}
                 provider={db.providers[0]}
                 location={db.locations[0]}
                 endCallback={null} />
    </BrowserRouter>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Snapshots',()=>{
  var tree;
  var instance;

  beforeEach(() => {
    let db = GetTestDb();
    let students = [
      new Student(0, null),
    ];
    students.forEach(student => {
      student.goals = [new Goal(0)];
    });

    tree = create(
      <BrowserRouter>
        <DataEntry active={true}
                   students={students}
                   provider={db.providers[0]}
                   location={db.locations[0]}
                   endCallback={null} />
      </BrowserRouter>
    )
    instance = tree.root.findByType(DataEntry)._fiber.stateNode;
  });

  afterEach(() => {
    tree = null;
    instance = null;
  });

  test('Initial data entry', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  })

  test('New student row when have 2 students', () => {
    let students = [
      new Student(0, null),
      new Student(1, null),
    ];

    instance.setState({
      students: {students}
    })
    // expect(instance.state.students.length).toBe(2);
    expect(tree.toJSON()).toMatchSnapshot();
  })

  test('New goal when have 2 goals', () => {
    let students = [
      new Student(0, null),
    ];
    students.forEach(student => {
      student.goals.push(new Goal(1));
    });

    instance.setState({
      students: {students}
    })
    // expect(instance.state.students[0].goals.length).toBe(2);
    expect(tree.toJSON()).toMatchSnapshot();
  })
})
