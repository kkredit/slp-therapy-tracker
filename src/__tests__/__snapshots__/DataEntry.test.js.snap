// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots Initial data entry 1`] = `
<div
  className="container"
  id="dataContainer"
>
  <div
    className="row"
  >
    <div
      className="col-1"
    />
    <div
      className="col-2"
    >
      <h3
        className="nameLocation"
      >
         
        sally
         
        foreman
         
      </h3>
    </div>
    <div
      className="col-2"
    >
      <h3
        className="nameLocation"
      >
         
        testroom1
         
      </h3>
    </div>
    <div
      className="col-4"
    />
    <div
      className="col-1"
    >
      <button
        className="endButton btn btn-outline-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-trash"
        />
         Discard
      </button>
    </div>
    <div
      className="col-1"
    >
      <button
        className="endButton btn btn-outline-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-save"
        />
         Submit
      </button>
    </div>
    <div
      className="col-1"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-1"
    >
      <h3
        className="dataColHdr"
      >
         Students 
      </h3>
    </div>
    <div
      className="col-1"
    />
    <div
      className="col-9"
    >
      <h3
        className="dataColHdr"
      >
         Goals 
      </h3>
    </div>
    <div
      className="col-1"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-1"
    >
      <h4
        className="studentName"
      >
         Student 
        1
         
      </h4>
    </div>
    <div
      className="col-11"
    >
      <div
        className="goalsCarousel carousel slide"
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="carousel-inner"
        >
          <div
            className="carousel-item active carousel-item"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <h4
                    className="goalNum"
                  >
                     Goal 
                    1
                     
                  </h4>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col"
                />
                <div
                  className="col-3"
                >
                  <button
                    className="goalButton btn btn-light btn-lg"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fa fa-3x fa-check"
                    />
                  </button>
                </div>
                <div
                  className="col-3"
                >
                  <button
                    className="goalButton btn btn-light btn-lg"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fa fa-3x fa-circle-o"
                    />
                  </button>
                </div>
                <div
                  className="col-3"
                >
                  <button
                    className="goalButton btn btn-light btn-lg"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fa fa-3x fa-times"
                    />
                  </button>
                </div>
                <div
                  className="col"
                />
              </div>
            </div>
          </div>
        </div>
        <a
          className="carousel-control-prev"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <i
            className="fa fa-3x fa-angle-left carouselDirections"
          />
          <span
            className="sr-only"
          >
            Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <i
            className="fa fa-3x fa-angle-right carouselDirections"
          />
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshots New goal when have 2 goals 1`] = `
<div
  className="container"
  id="dataContainer"
>
  <div
    className="row"
  >
    <div
      className="col-1"
    />
    <div
      className="col-2"
    >
      <h3
        className="nameLocation"
      >
         
        sally
         
        foreman
         
      </h3>
    </div>
    <div
      className="col-2"
    >
      <h3
        className="nameLocation"
      >
         
        testroom1
         
      </h3>
    </div>
    <div
      className="col-4"
    />
    <div
      className="col-1"
    >
      <button
        className="endButton btn btn-outline-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-trash"
        />
         Discard
      </button>
    </div>
    <div
      className="col-1"
    >
      <button
        className="endButton btn btn-outline-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-save"
        />
         Submit
      </button>
    </div>
    <div
      className="col-1"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-1"
    >
      <h3
        className="dataColHdr"
      >
         Students 
      </h3>
    </div>
    <div
      className="col-1"
    />
    <div
      className="col-9"
    >
      <h3
        className="dataColHdr"
      >
         Goals 
      </h3>
    </div>
    <div
      className="col-1"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-1"
    >
      <h4
        className="studentName"
      >
         Student 
        1
         
      </h4>
    </div>
    <div
      className="col-11"
    >
      <div
        className="goalsCarousel carousel slide"
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="carousel-inner"
        >
          <div
            className="carousel-item active carousel-item"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <h4
                    className="goalNum"
                  >
                     Goal 
                    1
                     
                  </h4>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col"
                />
                <div
                  className="col-3"
                >
                  <button
                    className="goalButton btn btn-light btn-lg"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fa fa-3x fa-check"
                    />
                  </button>
                </div>
                <div
                  className="col-3"
                >
                  <button
                    className="goalButton btn btn-light btn-lg"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fa fa-3x fa-circle-o"
                    />
                  </button>
                </div>
                <div
                  className="col-3"
                >
                  <button
                    className="goalButton btn btn-light btn-lg"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fa fa-3x fa-times"
                    />
                  </button>
                </div>
                <div
                  className="col"
                />
              </div>
            </div>
          </div>
        </div>
        <a
          className="carousel-control-prev"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <i
            className="fa fa-3x fa-angle-left carouselDirections"
          />
          <span
            className="sr-only"
          >
            Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <i
            className="fa fa-3x fa-angle-right carouselDirections"
          />
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshots New student row when have 2 students 1`] = `
<div
  className="container"
  id="dataContainer"
>
  <div
    className="row"
  >
    <div
      className="col-1"
    />
    <div
      className="col-2"
    >
      <h3
        className="nameLocation"
      >
         
        sally
         
        foreman
         
      </h3>
    </div>
    <div
      className="col-2"
    >
      <h3
        className="nameLocation"
      >
         
        testroom1
         
      </h3>
    </div>
    <div
      className="col-4"
    />
    <div
      className="col-1"
    >
      <button
        className="endButton btn btn-outline-danger"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-trash"
        />
         Discard
      </button>
    </div>
    <div
      className="col-1"
    >
      <button
        className="endButton btn btn-outline-success"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-save"
        />
         Submit
      </button>
    </div>
    <div
      className="col-1"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-1"
    >
      <h3
        className="dataColHdr"
      >
         Students 
      </h3>
    </div>
    <div
      className="col-1"
    />
    <div
      className="col-9"
    >
      <h3
        className="dataColHdr"
      >
         Goals 
      </h3>
    </div>
    <div
      className="col-1"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-1"
    >
      <h4
        className="studentName"
      >
         Student 
        1
         
      </h4>
    </div>
    <div
      className="col-11"
    >
      <div
        className="goalsCarousel carousel slide"
        onKeyDown={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="carousel-inner"
        >
          <div
            className="carousel-item active carousel-item"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <h4
                    className="goalNum"
                  >
                     Goal 
                    1
                     
                  </h4>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col"
                />
                <div
                  className="col-3"
                >
                  <button
                    className="goalButton btn btn-light btn-lg"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fa fa-3x fa-check"
                    />
                  </button>
                </div>
                <div
                  className="col-3"
                >
                  <button
                    className="goalButton btn btn-light btn-lg"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fa fa-3x fa-circle-o"
                    />
                  </button>
                </div>
                <div
                  className="col-3"
                >
                  <button
                    className="goalButton btn btn-light btn-lg"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <i
                      className="fa fa-3x fa-times"
                    />
                  </button>
                </div>
                <div
                  className="col"
                />
              </div>
            </div>
          </div>
        </div>
        <a
          className="carousel-control-prev"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <i
            className="fa fa-3x fa-angle-left carouselDirections"
          />
          <span
            className="sr-only"
          >
            Previous
          </span>
        </a>
        <a
          className="carousel-control-next"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          <i
            className="fa fa-3x fa-angle-right carouselDirections"
          />
          <span
            className="sr-only"
          >
            Next
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
